File: Backend/src/main/java/com/luckyseven/backend/domain/budget/dto/BudgetCreateRequest.java
package com.luckyseven.backend.domain.budget.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.luckyseven.backend.domain.budget.entity.CurrencyCode;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import java.math.BigDecimal;
import lombok.Getter;
import lombok.experimental.SuperBuilder;
import lombok.extern.jackson.Jacksonized;

public record BudgetCreateRequest (
    @NotNull
    @JsonProperty("totalAmount")
    @DecimalMin(value = "0.0", inclusive = false)
    BigDecimal totalAmount,

    @NotNull
    @JsonProperty("isExchanged")
    Boolean isExchanged,

    @JsonProperty("exchangeRate")
    @DecimalMin(value = "0.0", inclusive = false)
    BigDecimal exchangeRate,

    @NotNull
    @JsonProperty("foreignCurrency")
    CurrencyCode foreignCurrency
) { }

File: Backend/src/main/java/com/luckyseven/backend/domain/budget/dto/BudgetCreateRequest.kt
package com.luckyseven.backend.domain.budget.dto

import com.fasterxml.jackson.annotation.JsonProperty
import com.luckyseven.backend.domain.budget.entity.CurrencyCode
import jakarta.validation.constraints.DecimalMin
import jakarta.validation.constraints.NotNull
import java.math.BigDecimal

data class BudgetCreateRequest(

    @field:JsonProperty("totalAmount")
    @field:NotNull
    @field:DecimalMin(value = "0.0", inclusive = false)
    val totalAmount: BigDecimal,

    @field:JsonProperty("isExchanged")
    @field:NotNull
    val isExchanged: Boolean,

    @field:JsonProperty("exchangeRate")
    @field:DecimalMin(value = "0.0", inclusive = false)
    val exchangeRate: BigDecimal?,

    @field:JsonProperty("foreignCurrency")
    @field:NotNull
    val foreignCurrency: CurrencyCode
)
